src/wfs/src/main/java/org/geoserver/wfs/json/GeoJSONGetFeatureResponse.java

```diff
@@ -417,9 +420,19 @@ private int getFeatureCountFromWFS11Request(Operation operation, WFSInfo wfs)
             FeatureSource<? extends FeatureType, ? extends Feature> source = meta.getFeatureSource(
                     null, null);
             Filter filter = query.getFilter();
+            
             if (filter == null) {
                 filter = Filter.INCLUDE;
             }
+            else if (filter instanceof org.geotools.filter.spatial.AbstractPreparedGeometryFilter){
```

Comment:
This makes the bug easy to reproduce with a filter like "myatt = 5 && bbox(the_geom, x1,y,1,x2,y2)"
There is no need to check for the type of root filter, the visitor will scan the entire filter and reproject every spatial bit in it.